measureme/src/profiler.rs

```diff
@@ -111,4 +113,46 @@ impl<S: SerializationSink> Profiler<S> {
                 bytes.copy_from_slice(raw_event_bytes);
             });
     }
+
+    /// Creates a "start" event and returns a `TimingGuard` that will create
+    /// the corresponding "end" event when it is dropped.
+    pub fn start_event<'a>(
```

Comment:
Is there a better name for this? `start_recording` maybe?

Reply:
Is `start_recording_interval_event` too wordy?

Reply:
`start_recording` implies to me that if you don't call it, any calls to `record_event` will not be recorded.

I think this is fine for now.

Reply:
We could eventually split the api more and hide `record_event` and just expose two different methods `record_duration` and `record_instantaneous`.

Reply:
I changed it to `start_recording_interval_event` just now, and plan to later replace `record_event` with `record_instant_event`. Does that work for you?

Reply:
Yeah, that's fine.